tests/integration/tests/test_nvidia_gpu_operator.py

```diff
@@ -0,0 +1,255 @@
+
+NVIDIA_GPU_OPERATOR_HELM_CHART_REPO = "https://helm.ngc.nvidia.com/nvidia"
+
+# Mapping between the versions of the Nvidia `gpu-operator` and
+# the host versions of Ubuntu they support.
+# Because the `nvidia-driver-daemonset` pod included in the `gpu-operator`
+# includes kernel drivers, its container image's release lifecycle is
+# strictly tied to the version of Ubuntu on the host.
+# https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/platform-support.html
+NVIDIA_GPU_OPERATOR_SUPPORTED_UBUNTU_VERSIONS = {"v24.9.1": ["20.04", "22.04"]}
+
+NVIDIA_KERNEL_MODULE_NAMES = ["nvidia", "nvidia_uvm", "nvidia_modeset"]
+
+# Lifted 1:1 from:
+# https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html#cuda-vectoradd
+NVIDIA_CUDA_VECTOR_ADDITION_TEST_POD_NAME = "cuda-vectoradd"
+NVIDIA_CUDA_VECTOR_ADDITION_TEST_POD_SPEC = f"""
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {NVIDIA_CUDA_VECTOR_ADDITION_TEST_POD_NAME}
+spec:
+  restartPolicy: OnFailure
+  containers:
+  - name: cuda-vectoradd
+    image: "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda11.7.1-ubuntu20.04"
+    resources:
+      limits:
+        nvidia.com/gpu: 1
+"""[
+    1:
+]
+
+
+def _check_nvidia_gpu_present(instance: harness.Instance) -> bool:
+    """Checks whether at least one Nvidia GPU is available
+    by exec-ing `lspci` on the target instance."""
+    proc = instance.exec(["lspci", "-k"], capture_output=True)
+
+    for line in proc.stdout.split(b"\n"):
+        if b"NVIDIA Corporation" in line:
+            LOG.info(f"Found NVIDIA GPU in lspci output: {line}")
+            return True
+
+    LOG.info(f"Failed to find NVIDIA GPU in lspci output: {proc.stdout}")
+    return False
+
+
+def _check_nvidia_drivers_loaded(instance: harness.Instance) -> Mapping[str, bool]:
+    """Ensures that Nvidia kernel medules are NOT loaded on
+    the given harness instance."""
+
+    proc = instance.exec(["lsmod"], capture_output=True)
+    modules_present = {m: False for m in NVIDIA_KERNEL_MODULE_NAMES}
+    for line in proc.stdout.split(b"\n"):
+        for mod in modules_present:
+            if line.startswith(mod.encode()):
+                modules_present[mod] = True
+
+    LOG.info(f"Located the following Nvidia kernel modules {modules_present}")
+    return modules_present
+
+
+def _get_harness_instance_os_release(instance: harness.Instance) -> str:
+    """Reads harness instance os series from LSB release."""
+    proc = instance.exec(["cat", "/etc/lsb-release"], capture_output=True)
+
+    release = None
+    var = "DISTRIB_RELEASE"
+    for line in proc.stdout.split(b"\n"):
+        line = line.decode()
+        if line.startswith(var):
+            release = line.lstrip(f"{var}=")
+            break
+
+    if release is None:
+        raise ValueError(
+            f"Failed to parse LSB release var '{release}' from LSB info: "
+            f"{proc.stdout}"
+        )
+
+    return release
+
+
+def wait_for_gpu_operator_daemonset(
```

Comment:
I was aware we already had some utils in github.com/canonical/k8s-test-harness, but I wanted to avoid importing it/copying their implementations directly ([the `wait_for_daemonset` from there was relying on `rollout status`](https://github.com/canonical/k8s-test-harness/blob/main/k8s_test_harness/util/k8s_util.py#L214-L215) which won't work unless you have `RollingUpdate`, for example).

Anyhow, I've moved it to `util.py` and made it decently generic (parametrized the expected pod ready count)